go/vt/vtgate/semantics/binder.go

```diff
@@ -85,7 +85,23 @@ func (b *binder) up(cursor *sqlparser.Cursor) error {
 			currScope.joinUsing[ident.Lowered()] = deps.direct
 		}
 		if len(node.Using) > 0 {
-			err := rewriteJoinUsing(currScope, node.Using, b.org)
+			err := rewriteJoinUsing(currScope, node, b.org)
+			if err != nil {
+				return err
+			}
+			err = sqlparser.Walk(func(node sqlparser.SQLNode) (kontinue bool, err error) {
+				column, isColumn := node.(*sqlparser.ColName)
+				if !isColumn {
+					return true, nil
+				}
+				deps, err := b.resolveColumn(column, currScope, false)
+				if err != nil {
+					return false, err
+				}
+				b.recursive[column] = deps.recursive
+				b.direct[column] = deps.direct
+				return true, nil
+			}, node.On)
```

Comment:
```
join_specification: {
    ON search_condition
  | USING (join_column_list)
}

join_column_list:
    column_name [, column_name] ...
```

`Using` can only have `column_name`, I think we can skip tree walking and can speed up the dependency assignment.

Reply:
Also `resolveColumn` call is not needed, when `Using` join spec is used, the column should belong to both left and right tableExpr. Probably can get that information directly and assign it.

Reply:
> `Using` can only have `column_name`, I think we can skip tree walking and can speed up the dependency assignment.

I am not sure I understand this correctly. I am walking on `node.On` which is a sqlparser expression. My thinking was that I wanted to catch all columns in this expression and assign dependency on it.